### PR TITLE
[zh] Update service.md to fix some wrongs.

### PR DESCRIPTION
**Pull request by yuandongx**
I think the ZH is incorrect translation about this line, so update it.
## EN:
` To learn about other ways to define Service endpoints, see Services without selectors.`
## ZH should be:
`要了解定义服务端点的其他方法，请参阅[不带选择算符的服务]`

### CODE REVIEW

**Comment by tengqm** [comment]
/hold

This page is out of sync, please resync and then fix the typo.

```
./scripts/lsync.sh content/zh/docs/concepts/services-networking/service.md
```

**Comment by yuandongx** [comment]
Sync .... WIP

File: content/zh/docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -339,16 +338,22 @@ Endpoint 切片提供了附加的属性和功能，这些属性和功能在
...
 ### 应用程序协议   {#application-protocol}
 
-{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+{{< feature-state for_k8s_version="v1.20" state="stable" >}}
 
 `appProtocol` 字段提供了一种为每个 Service 端口指定应用协议的方式。
-此字段的取值会被映射到对应的 Endpoints 和 EndpointSlices 资源。
-
+此字段的取值会被映射到对应的 Endpoints 和 EndpointSlices 对象。
+该字段遵循标准的 Kubernetes 标签语法。其值可以是 [IANA 标准服务名称]或以域名前缀的名称，
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:361]: 链接丢了。

File: content/zh/docs/concepts/services-networking/service.md [hunk 2]
```diff
@@ -513,23 +518,23 @@ IPVS代理模式基于类似于 iptables 模式的 netfilter 挂钩函数，
...
-- `dh`: destination hashing
-- `sh`: source hashing
-- `sed`: shortest expected delay
-- `nq`: never queue
+* `rr`: round-robin
+* `lc`: least connection (smallest number of open connections)
+* `dh`: destination hashing
+* `sh`: source hashing
+* `sed`: shortest expected delay
+* `nq`: never queue
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:532]:
```suggestion
* `rr`：轮替（Round-Robin）
* `lc`：最少链接（Least Connection），即打开链接数量最少者优先
* `dh`：目标地址哈希（Destination Hashing）
* `sh`：源地址哈希（Source Hashing）
* `sed`：最短预期延迟（Shortest Expected Delay）
* `nq`：从不排队（Never Queue）
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 3]
```diff
@@ -715,35 +720,35 @@ Services by their DNS name.
...
 Pods in other Namespaces must qualify the name as `my-service.my-ns`. These names
 will resolve to the cluster IP assigned for the Service.
 -->
-例如，如果你在 Kubernetes 命名空间 `"my-ns"` 中有一个名为 `"my-service"` 的服务，
-则控制平面和 DNS 服务共同为 `"my-service.my-ns"` 创建 DNS 记录。
-`"my-ns"` 命名空间中的 Pod 应该能够通过简单地对 `my-service` 进行名称查找来找到它
-（`"my-service.my-ns"` 也可以工作）。
+例如，如果你在 Kubernetes 命名空间 `my-ns` 中有一个名为 `my-service` 的服务，
+则控制平面和 DNS 服务共同为 `my-service.my-ns` 创建 DNS 记录。
+`my-ns` 命名空间中的 Pod 应该能够通过简单地对 `my-service` 进行名称查找来找到它
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:739]:
```suggestion
`my-ns` 命名空间中的 Pod 应该能够通过简单地按名检索 `my-service` 来找到它
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 4]
```diff
@@ -922,15 +925,15 @@ to configure environments that are not fully supported by Kubernetes, or even
...
+and `.spec.clusterIP:spec.ports[*].port`. (If the `--nodeport-addresses` flag in kube-proxy is set, <NodeIP> would be filtered NodeIP(s).)
 
 For example:
 -->
 使用 NodePort 可以让你自由设置自己的负载均衡解决方案，配置 Kubernetes 不完全支持的环境，
 甚至直接暴露一个或多个节点的 IP。
 
 需要注意的是，Service 能够通过 `<NodeIP>:spec.ports[*].nodePort` 和
-`spec.clusterIp:spec.ports[*].port` 而对外可见。
+`spec.clusterIp:spec.ports[*].port` 而对外可见（如果 kube-proxy 的 `--nodeport-addresses` 参数被设置了， <NodeIP>将被过滤 NodeIP。）。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:942]: 折行

File: content/zh/docs/concepts/services-networking/service.md [hunk 5]
```diff
@@ -984,26 +987,18 @@ spec:
       port: 80
       targetPort: 9376
   clusterIP: 10.0.171.239
-  loadBalancerIP: 78.11.24.19
   type: LoadBalancer
 status:
   loadBalancer:
     ingress:
-      - ip: 146.148.47.155
+    - ip: 192.0.2.127
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1000]: 原来的缩进更清晰

File: content/zh/docs/concepts/services-networking/service.md [hunk 6]
```diff
@@ -1046,7 +1041,54 @@ securityGroupName。
...
+
+{{< feature-state for_k8s_version="v1.20" state="alpha" >}}
+
+By default, for LoadBalancer type of Services, when there is more than one port defined, all
+ports must have the same protocol, and the protocol must be one which is supported
+by the cloud provider.
+
+If the feature gate `MixedProtocolLBService` is enabled for the kube-apiserver it is allowed to use different protocols when there is more than one port defined.
+-->
+#### 混合协议类型的负载平衡器
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1053]: 负载均衡器

File: content/zh/docs/concepts/services-networking/service.md [hunk 7]
```diff
@@ -1046,7 +1041,54 @@ securityGroupName。
...
+
+{{< feature-state for_k8s_version="v1.20" state="alpha" >}}
+
+默认情况下，对于 LoadBalancer 类型的服务，当定义了多个端口时，所有
+端口必须具有相同的协议，并且该协议必须是受支持的协议由云提供商提供。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1058]:
```suggestion
端口必须具有相同的协议，并且该协议必须是受云提供商支持的协议。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 8]
```diff
@@ -1046,7 +1041,54 @@ securityGroupName。
...
+如果为 kube-apiserver 启用了 “MixedProtocolLBService” 功能，则当定义了多个端口时，允许使用不同的协议。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md]:
```suggestion
如果为 kube-apiserver 启用了 `MixedProtocolLBService` 特性门控，则当定义了多个端口时，允许使用不同的协议。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 9]
```diff
@@ -1046,7 +1041,54 @@ securityGroupName。
...
+
+<!--
+The set of protocols that can be used for LoadBalancer type of Services is still defined by the cloud provider.
+-->
+{{< note >}}
+可用于 LoadBalancer 类型服务的协议集仍然由云提供商定义。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1066]:
```suggestion
可用于 LoadBalancer 类型服务的协议集仍然由云提供商决定。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 10]
```diff
@@ -1046,7 +1041,54 @@ securityGroupName。
...
+is `true` and type LoadBalancer Services will continue to allocate node ports. If `spec.allocateLoadBalancerNodePorts`
+is set to `false` on an existing Service with allocated node ports, those node ports will NOT be de-allocated automatically.
+You must explicitly remove the `nodePorts` entry in every Service port to de-allocate those node ports.
+You must enable the `ServiceLBNodePortControl` feature gate to use this field.
+-->
+### 禁用负载均衡器节点端口分配 {#load-balancer-nodeport-allocation}
 
+{{< feature-state for_k8s_version="v1.20" state="alpha" >}}
+
+从 v1.20 版本开始， 你可以通过设置 `spec.allocateLoadBalancerNodePorts` 为 `false` 对类型=负载均衡器的服务进行禁用节点端口分配功能。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1085]:
```suggestion
从 v1.20 版本开始， 你可以通过设置 `spec.allocateLoadBalancerNodePorts` 为 `false` 
对类型为 LoadBalancer 的服务禁用节点端口分配。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 11]
```diff
@@ -1046,7 +1041,54 @@ securityGroupName。
...
+这仅适用于负载平衡器实现直接将流量路由到 Pod，而不是使用节点端口。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1086]:
```suggestion
这仅适用于直接将流量路由到 Pod而不是使用节点端口的负载均衡器实现。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 12]
```diff
@@ -1046,7 +1039,55 @@ securityGroupName。
...
+You must explicitly remove the `nodePorts` entry in every Service port to de-allocate those node ports.
+You must enable the `ServiceLBNodePortControl` feature gate to use this field.
+-->
+### 禁用负载均衡器节点端口分配 {#load-balancer-nodeport-allocation}
+
+{{< feature-state for_k8s_version="v1.20" state="alpha" >}}
+
+从 v1.20 版本开始， 你可以通过设置 `spec.allocateLoadBalancerNodePorts` 为 `false` 对类型=负载均衡器的服务进行禁用节点端口分配功能。
+这仅适用于负载平衡器实现直接将流量路由到 Pod，而不是使用节点端口。
+默认情况下，`spec.allocateLoadBalancerNodePorts` 为 `true`, 负载均衡服务将持续分配端口。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1087]:
```suggestion
默认情况下，`spec.allocateLoadBalancerNodePorts` 为 `true`，LoadBalancer 类型的服务继续分配节点端口。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 13]
```diff
@@ -1046,7 +1039,55 @@ securityGroupName。
...
+如果已存在的服务被分配有端口，将参数 `spec.allocateLoadBalancerNodePorts` 设置为 `false`，这些服务上已分配置的端口将不会被自动释放。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1088]:
```suggestion
如果现有服务已被分配节点端口，将参数 `spec.allocateLoadBalancerNodePorts` 设置为 `false` 时，
这些服务上已分配置的节点端口不会被自动释放。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 14]
```diff
@@ -1046,7 +1039,55 @@ securityGroupName。
...
+你必须显示的在每个服务端口中删除 `nodePorts`，以释放它们。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1089]:
```suggestion
你必须显式地在每个服务端口中删除 `nodePorts` 项以释放对应端口。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 15]
```diff
@@ -1046,7 +1039,55 @@ securityGroupName。
...
+要使用该字段，你必须启用 `ServiceLBNodePortControl` 特性功能。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1090]:
```suggestion
你必须启用 `ServiceLBNodePortControl` 特性门控才能使用该字段。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 16]
```diff
@@ -1055,16 +1096,16 @@ In a mixed environment it is sometimes necessary to route traffic from Services
 
 In a split-horizon DNS environment you would need two Services to be able to route both external and internal traffic to your endpoints.
 
-You can achieve this by adding one the following annotations to a Service.
-The annotation to add depends on the cloud Service provider you're using.
+To set an internal load balancer, add one of the following annotations to your Service
+depending on the cloud Service provider you're using.
 -->
-#### 内部负载均衡器   {#internal-load-balancer}
+### 内部负载均衡器
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1102]: 这一行不要动

File: content/zh/docs/concepts/services-networking/service.md [hunk 17]
```diff
@@ -1055,16 +1096,16 @@ In a mixed environment it is sometimes necessary to route traffic from Services
...
 
 在混合环境中，有时有必要在同一(虚拟)网络地址块内路由来自服务的流量。
 
 在水平分割 DNS 环境中，你需要两个服务才能将内部和外部流量都路由到你的端点（Endpoints）。
-你可以通过向服务添加以下注解之一来实现此目的。
-要添加的注解取决于你使用的云服务提供商。
+
+如要设置内部负载均衡器，请根据你的云运营商添加以下注释之一。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1108]:
```suggestion
如要设置内部负载均衡器，请根据你所使用的云运营商，为服务添加以下注解之一。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 18]
```diff
@@ -1240,9 +1281,11 @@ In the above example, if the Service contained three ports, `80`, `443`, and
 `8443`, then `443` and `8443` would use the SSL certificate, but `80` would just
 be proxied HTTP.
 
-From Kubernetes v1.9 onwards you can use [predefined AWS SSL policies](http://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-security-policy-table.html) with HTTPS or SSL listeners for your Services.
+From Kubernetes v1.9 onwards you can use [predefined AWS SSL policies](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-security-policy-table.html) with HTTPS or SSL listeners for your Services.
 To see which policies are available for use, you can use the `aws` command line tool:
 -->
+在上述的例子中，如果服务包含 `80`、`443` 和 `8443` 三个端口， 那么 `443` 和 `8443` 将使用SSL证书，而 `80` 将仅仅是代理 HTPP。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1287]:
```suggestion
在上例中，如果服务包含 `80`、`443` 和 `8443` 三个端口， 那么 `443` 和 `8443` 将使用 SSL 证书，
而 `80` 端口将仅仅转发 HTTP 数据包。
```

File: content/zh/docs/concepts/services-networking/service.md [hunk 19]
```diff
@@ -1555,6 +1560,78 @@ in those modified security groups.
...
+        service.kubernetes.io/qcloud-loadbalancer-internet-max-bandwidth-out: "10"
+
+        # When this annotation is set，the loadbalancers will only register nodes
+        # with pod running on it, otherwise all nodes will be registered.
+        service.kubernetes.io/local-svc-only-bind-node-with-pod: true
+```
+-->
+#### 腾讯 Kubernetes 引擎（TKE）上的 CLB 注释
+
+以下是在 TKE 上管理云负载均衡器的注释。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1602]: 注解

File: content/zh/docs/concepts/services-networking/service.md [hunk 20]
```diff
@@ -1555,6 +1560,78 @@ in those modified security groups.
...
+
+```yaml
+    metadata:
+      name: my-service
+      annotations:
+        # Bind Loadbalancers with specified nodes
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1608]: YAML 中的注解也要翻译

File: content/zh/docs/concepts/services-networking/service.md [hunk 21]
```diff
@@ -1876,6 +1953,67 @@ You can use UDP for most Services. For type=LoadBalancer Services, UDP support
...
+{{< note >}}
+SCTP is not supported on Windows based nodes.
+{{< /note >}}
+
+##### Userspace kube-proxy {#caveat-sctp-kube-proxy-userspace}
+
+{{< warning >}}
+The kube-proxy does not support the management of SCTP associations when it is in userspace mode.
+{{< /warning >}}
+-->
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1987]: 请将以上大段分开翻译，现在无法 review

File: content/zh/docs/concepts/services-networking/service.md (hunk 1 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:361]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 2 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:532]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 3 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:739]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 4 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:942]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 5 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1000]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 6 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1053]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 7 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1058]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 8 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 9 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1066]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 10 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1085]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 11 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1086]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 12 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1087]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 13 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1088]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 14 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1089]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 15 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1090]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 16 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md:1102]: Done.

File: content/zh/docs/concepts/services-networking/service.md [hunk 22]
```diff
@@ -1555,6 +1567,78 @@ in those modified security groups.
...
+#### 腾讯 Kubernetes 引擎（TKE）上的 CLB 注解
+
+以下是在 TKE 上管理云负载均衡器的注解。
+
+```yaml
+    metadata:
+      name: my-service
+      annotations:
+		# 绑定负载均衡器到指定的节点。
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1615]: 这一行有问题

File: content/zh/docs/concepts/services-networking/service.md [hunk 23]
```diff
@@ -1555,6 +1567,78 @@ in those modified security groups.
...
+        service.kubernetes.io/loadbalance-type: xxxxx
+
+        # 指定公用网络带宽计费方法。
+        # 可用参数: TRAFFIC_POSTPAID_BY_HOUR(bill-by-traffic) 和 BANDWIDTH_POSTPAID_BY_HOUR (bill-by-bandwidth).
+        service.kubernetes.io/qcloud-loadbalancer-internet-charge-type: xxxxxx
+
+        # 指定带宽参数 (取值范围： [1,2000] Mbps).
+        service.kubernetes.io/qcloud-loadbalancer-internet-max-bandwidth-out: "10"
+
+		# 当设置该注解时，负载平衡器将只注册正在运行 Pod 的节点，
```
Comment (tengqm) [inline content/zh/docs/concepts/services-networking/service.md:1638]: 同上

File: content/zh/docs/concepts/services-networking/service.md (hunk 22 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md]: Done.

File: content/zh/docs/concepts/services-networking/service.md (hunk 23 above)
Comment (yuandongx) [reply content/zh/docs/concepts/services-networking/service.md]: Thanks, done.

**Comment by yuandongx** [comment]
cc @howieyuen

**Comment by yuandongx** [comment]
@tengqm 
It seems that two `lgtm` labels doesn't merge the PR.
If a `approve` is necessary ?

**Comment by tengqm** [comment]
@yuandongx pls be patient.

**Review by tanjunchen — COMMENTED** [review]
/approve

**Pull request merged**